raphtory/src/algorithms/components/out_components.rs

```diff
@@ -94,12 +96,88 @@ where
     AlgorithmResult::new(graph.clone(), "Out Components", results_type, res)
 }
 
+/// Computes the out-component of a given node in the graph
+///
+/// # Arguments
+///
+/// * `node` - The node whose out-component we wish to calculate
+///
+/// Returns:
+///
+/// A Vec containing the Nodes within the given nodes out-component
+///
+pub fn out_component<'graph, G: GraphViewOps<'graph>>(node: NodeView<G>) -> Vec<NodeView<G>> {
+    let mut out_components = HashSet::new();
+    let mut to_check_stack = Vec::new();
+    node.out_neighbours().iter().for_each(|node| {
+        let id = node.node;
+        out_components.insert(id);
```

Comment:
Same optimisation as for the `in_component`, only push to the stack if the node is not already in the set.